acstools/acszpt.py

```diff
@@ -0,0 +1,329 @@
+
+    def _submit_request(self):
+        """ Submit a request to the ACS Zeropoint Calculator
+
+        If an exception is raised during the request an error message is
+        printed. Otherwise, the response is saved in the corresponding
+        attribute.
+        """
+        try:
+            self._response = requests.get(self._url)
+        except requests.exceptions.RequestException as e:
+            print(e)
+            print('-'*79)
+            print('The query failed! Check your input args and if the error '
+                  'persists submit a ticket to the ACS Help Desk at '
+                  'hsthelp.stsci.edu with the error message displayed above.')
+            self._failed = True
+
+    def _parse_response(self):
+        """ Parse the html response returned from the HTML request to find all
+        tables on the webpage.
+
+        Using bs4, find all the <tb> </tb> tags present in the response. Save
+        the results in an OrderedDict() so when parsing occurs we get stable
+        results.
+
+        """
+        soup = BeautifulSoup(
+            self._response.content.decode(self._response.encoding),
+            'html.parser'
+        )
+        # Grab all elements in the table returned by the ZPT calc.
+        td = soup.find_all('td')
+        for j in range(6):
+            for i, val in enumerate(td[j::6]):
+                if not i:
+                    self._zpts[val.text.split('[')[0].strip()] = []
+                else:
+                    self._zpts[list(self._zpts.keys())[j]].append(val.text)
+
+    def _format_results(self):
+        """ Format the results into an astropy.table.Table with corresponding
+        units
+
+        Returns
+        -------
+        astropy.table.Table object with the zeropoints for the supplied date
+        and detector
+        """
+        data_types = []
+        data_units = []
+        for i, key in enumerate(self._zpts.keys()):
+            if key.lower() == 'filter':
+                data_types.append(str)
+                data_units.append(u.dimensionless_unscaled)
+
+            elif key.lower() == 'photplam':
+                data_types.append(float)
+                data_units.append(u.angstrom)
+
+            elif key.lower() == 'photflam':
+                data_types.append(float)
+                data_units.append(u.erg / u.cm**2 / u.second / u.angstrom)
+
+            elif key.lower() == 'stmag':
+                data_types.append(float)
+                data_units.append(u.STmag)
+
+            elif key.lower() == 'vegamag':
+                data_types.append(float)
+                data_units.append(u.mag)
+
+            elif key.lower() == 'abmag':
+                data_types.append(float)
+                data_units.append(u.ABmag)
+
+        tab = Table(list(self._zpts.values()),
+                    names=list(self._zpts.keys()),
+                    dtype=data_types)
+
+        # Loop through each column and set the appropriate units
```

Comment:
Is this necessary? Does `astropy.table.QTable` handle this for you?

Reply:
Which part are you referring to? The long block of `if/elif` statements is required to specify the units and datatypes for each column. The only other option is to hard code a list with those values, but we run the risk of the hard coded lists not being in the same order as the columns, i.e. we could potentially say the `FILTER` column is comprised of `float` with `u.ABmag` or something.